ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs

```diff
@@ -726,30 +732,15 @@ protocolInfoCardano paramsCardano
         , conwayMaxTxCapacityOverrides = maxTxCapacityOverridesConway
         } = paramsConway
 
-    ProtocolTransitionParamsByronToShelley {
-          transitionByronToShelleyTranslationContext = transCtxtShelley
-        , transitionByronToShelleyTrigger            = triggerHardForkShelley
-        } = transitionParamsByronToShelley
-    ProtocolTransitionParamsIntraShelley {
-          transitionIntraShelleyTranslationContext = ()
-        , transitionIntraShelleyTrigger            = triggerHardForkAllegra
-        } = transitionParamsShelleyToAllegra
-    ProtocolTransitionParamsIntraShelley {
-          transitionIntraShelleyTranslationContext = ()
-        , transitionIntraShelleyTrigger            = triggerHardForkMary
-        } = transitionParamsAllegraToMary
-    ProtocolTransitionParamsIntraShelley {
-          transitionIntraShelleyTranslationContext = transCtxtAlonzo
-        , transitionIntraShelleyTrigger            = triggerHardForkAlonzo
-        } = transitionParamsMaryToAlonzo
-    ProtocolTransitionParamsIntraShelley {
-          transitionIntraShelleyTranslationContext = transCtxtBabbage
-        , transitionIntraShelleyTrigger            = triggerHardForkBabbage
-        } = transitionParamsAlonzoToBabbage
-    ProtocolTransitionParamsIntraShelley {
-          transitionIntraShelleyTranslationContext = transCtxtConway
-        , transitionIntraShelleyTrigger            = triggerHardForkConway
-        } = transitionParamsBabbageToConway
+    transitionConfigs :: NP WrapTransitionConfig (CardanoShelleyEras c)
+    transitionConfigs@(  WrapTransitionConfig transitionConfigShelley
```

Comment:
Would it make sense to have a pattern similar to `CardanoHardForkTriggers'` here as well?

Reply:
Why do you need `transitionConfigs` at all? 

You ever only need `transitionConfig` for the latest era, since it will contain `transitionConfig` for all previous eras

Reply:
> Would it make sense to have a pattern similar to CardanoHardForkTriggers' here as well?

We could do that, but it would be only used here, and it is not a type that third parties will use.

> Why do you need `transitionConfigs` at all?
> 
> You ever only need `transitionConfig` for the latest era, since it will contain `transitionConfig` for all previous eras

Sure, but you need the `TransitionConfig` of a specific era in order to register the initial funds/staking for that era, which is where we use `transitionConfigs`:

https://github.com/input-output-hk/ouroboros-consensus/blob/29c591b9eb6cd2c2bdb803ad3c63cdda7f6f3ed1/ouroboros-consensus-cardano/src/ouroboros-consensus-cardano/Ouroboros/Consensus/Cardano/Node.hs#L1031

Eg it might be the case that people configure their node to skip ahead directly to say Allegra *and* they specify initial funds/staking. In that case, we need to register those, but as we don't statically know to which era they skip, we need to keep an `NP` of corresponding functions around.

We could also create this `NP` more manually, without the helper classes at the bottom of this file; the current state prevents boilerplate mistakes and automatically adapts to new eras, but it requires a bit of typeclass juggling.

Reply:
> And you can always get the transitionConfig for that era from the latest one

Sure, that is what we are doing here (a bit more manually since fdc687428b61a27bec27848e01c6e1b5bd38d270).

> you don't need to keep them all around.

I am not exactly sure what you mean by that; for each era, we need a registration function (we don't know which, if any, we have to call), and each of those require a `TransitionConfig` for that era. So each of these functions closures has to refer to one of those `TransitionConfig`s, so they are kept around until initial registration has been performed. Does that make sense or were you talking about sth different?

Reply:
I am always in favor of more readable code, even if it results in a little bit more boiler plate. That's why I pushed back on this general solution that was removed in: https://github.com/input-output-hk/ouroboros-consensus/commit/fdc687428b61a27bec27848e01c6e1b5bd38d270

That being said, it is not my place to tell you how to write code, so feel free to choose the solution that you prefer.

Reply:
No, thanks for raising that point, it is especially useful to get feedback on the parts of our API that get used by other teams/projects! What I like about these "general" solutions is that they eliminate potentially error-prone boilerplate, but I agree that it wasn't really worth it here in the scope of this PR.